Dart/src/com/jetbrains/lang/dart/ide/generation/DartGenerateToStringHandler.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2000-2016 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.jetbrains.lang.dart.ide.generation;
+
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.util.Condition;
+import com.intellij.openapi.util.Pair;
+import com.intellij.psi.PsiFile;
+import com.intellij.util.containers.ContainerUtil;
+import com.jetbrains.lang.dart.DartBundle;
+import com.jetbrains.lang.dart.DartComponentType;
+import com.jetbrains.lang.dart.psi.DartClass;
+import com.jetbrains.lang.dart.psi.DartComponent;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.Map;
+
+public class DartGenerateToStringHandler extends BaseDartGenerateHandler {
+  @Override
+  protected String getTitle() {
+    return DartBundle.message("dart.generate.toString");
+  }
+
+  @Override
+  protected BaseCreateMethodsFix createFix(DartClass dartClass) {
+    return new CreateToStringFix(dartClass);
+  }
+
+  @Override
+  protected void collectCandidates(DartClass aClass, List<DartComponent> candidates) {
+    super.collectCandidates(aClass, candidates);
+  }
+
+  @Override
+  protected void collectCandidates(Map<Pair<String, Boolean>, DartComponent> classMembersMap,
+                                   Map<Pair<String, Boolean>, DartComponent> superClassesMembersMap,
+                                   Map<Pair<String, Boolean>, DartComponent> superInterfacesMembersMap,
+                                   List<DartComponent> candidates) {
+    candidates.addAll(ContainerUtil.findAll(classMembersMap.values(), new Condition<DartComponent>() {
+      @Override
+      public boolean value(DartComponent component) {
+        return DartComponentType.typeOf(component) == DartComponentType.FIELD;
+      }
+    }));
+  }
+
+  @Override
+  public boolean isValidFor(@NotNull Editor editor, @NotNull PsiFile file) {
+    return super.isValidFor(editor, file);
+  }
```

Comment:
Why method overridden with no additional logic?
If you decide to disable action if `toString()` is already declared in the current class you can implement this check here.